docs/modules/vector-search/pages/vector-collections.adoc

```diff
@@ -0,0 +1,498 @@
+|===
+|Name|Description| Score definition
+
+|EUCLIDEAN
+|euclidean distance
+|1 / (1 + squareDistance(v1, v2))
+
+|COSINE
+|cosine of the angle between the vectors
+|(1 + cos(v1, v2)) / 2
+
+|DOT
+|dot product of the vectors
+|(1 + dotProduct(v1, v2)) / 2
+
+|===
+
+WARNING: The recommended method for computing cosine similarity is to normalize all vectors to unit length and utilize the DOT metric instead.
+
+
+Configuration example:
+
+[tabs]
+====
+XML::
++
+--
+[source,xml]
+----
+<hazelcast>
+    <vector-collection name="vector-collection-name">
+        <indexes>
+            <index name="index-1">
+                <dimension>2</dimension>
+                <metric>DOT</metric>
+            </index>
+            <index name="index-2">
+                <dimension>2</dimension>
+                <metric>DOT</metric>
+                <max-degree>10</max-degree>
+                <ef-construction>20</ef-construction>
+                <use-deduplication>false</use-deduplication>
+            </index>
+        </indexes>
+    </vector-collection>
+</hazelcast>
+----
+--
+YAML::
++
+--
+[source,yaml]
+----
+hazelcast:
+  vector-collection:
+    vector-collection-name:
+      indexes:
+        - name: index-1
+          dimension: 2
+          metric: DOT
+        - name: index-2
+          dimension: 2
+          metric: DOT
+          max-degree: 10
+          ef-construction: 20
+          use-deduplication: false
+----
+--
+Java::
++
+--
+[source,java]
+----
+Config config = new Config();
+VectorCollectionConfig collectionConfig = new VectorCollectionConfig("vector-collection-name")
+    .addVectorIndexConfig(
+            new VectorIndexConfig()
+                .setName("index-1")
+                .setDimension(2)
+                .setMetric(Metric.DOT)
+    ).addVectorIndexConfig(
+            new VectorIndexConfig()
+                .setName("index-2")
+                .setDimension(2)
+                .setMetric(Metric.DOT)
+                .setMaxDegree(10)
+                .setEfConstruction(20)
+                .setUseDeduplication(false)
+    );
+config.addVectorCollectionConfig(collectionConfig);
+----
+--
+====
+
+== Manage collection
+
+=== Create collection
+
+To create a vector collection, use the static method `getCollection(HazelcastInstance instance, VectorCollectionConfig collectionConfig)` of the `VectorCollection` class.
+
+Example:
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection vectorCollection = VectorCollection.getCollection(hazelcastInstance, vectorCollectionConfig);
+----
+--
+====
+
+=== Get existing vector collection
+
+To get a vector collection, use the static method `getCollection(HazelcastInstance instance, String collectionName)` of the `VectorCollection` class.
+
+Example:
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+----
+--
+====
+
+=== Optimize collection
```

Comment:
I'd move optimization after more ordinary operations

Reply:
Optimize method is more part of managing the collection and not managing data.

Reply:
but it is not the first thing you do after creating a collection. IMO more logical and easier to read order is: create/get collection, put/get data, search, optimize. Putting create and optimize together for the sake of grouping as "manage" does not justify a bit awkward reading order.

Reply:
7e1b5c3452fd6c95ac9cd6d02014415617138c44